packages/common/src/utils/filters.ts

```diff
@@ -104,15 +104,21 @@ export const filterableDimensionsOnly = (
 ): FilterableDimension[] => dimensions.filter(isFilterableDimension);
 
 export const getFilterTypeFromItem = (item: FilterableItem): FilterType => {
-    if (isTableCalculation(item)) {
+    if (isTableCalculation(item) && item.type === undefined) {
         return FilterType.NUMBER;
     }
 
     const { type } = item;
 
+    if (type === undefined) {
+        // Type check for TableCalculationType
+        return FilterType.NUMBER;
+    }
+
```

Comment:
[nitpick]

wouldn't this be the same as just doing:

```suggestion
    const { type } = item;

    if (type === undefined) {
        // Type check for TableCalculationType
        return FilterType.NUMBER;
    }
```